hyperdx-helm/templates/otel-collector-deployment.yaml

```diff
@@ -0,0 +1,62 @@
+{{- if .Values.otel.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "hdx-oss.fullname" . }}-otel-collector
+  labels:
+    {{- include "hdx-oss.labels" . | nindent 4 }}
+    app: otel-collector
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "hdx-oss.selectorLabels" . | nindent 6 }}
+      app: otel-collector
+  template:
+    metadata:
+      labels:
+        {{- include "hdx-oss.selectorLabels" . | nindent 8 }}
+        app: otel-collector
+    spec:
+      containers:
+        - name: otel-collector
+          image: "{{ .Values.images.otelCollector.repository }}:{{ .Values.images.otelCollector.tag }}"
+          ports:
+            - containerPort: {{ .Values.otel.port }}
+            - containerPort: {{ .Values.otel.nativePort }}
+            - containerPort: {{ .Values.otel.grpcPort }}
+            - containerPort: {{ .Values.otel.httpPort }}
+            - containerPort: {{ .Values.otel.healthPort }}
+          env:
+            - name: CLICKHOUSE_SERVER_ENDPOINT
+              value: "{{ include "hdx-oss.fullname" . }}-clickhouse:{{ .Values.clickhouse.port }}"
```

Comment:
we need to change `clickhouse.port` to clickhouse.nativePort`. Also I realized we need to inject `CLICKHOUSE_USER` and  `CLICKHOUSE_PASSWORD` (with otelcollector user)

Reply:
Done